File: src/main/java/mk/majstor/service/OffersService.java
package mk.majstor.service;

import mk.majstor.model.listing.Listing;
import mk.majstor.model.offer.Offer;
import mk.majstor.model.offer.OfferDto;
import org.springframework.data.domain.Page;

public interface OffersService {

    Page<Offer> getAllOffersByListing(Long listingId, int pageNo, int pageSize);

    Page<Offer> getAllOffersByListingByUser(Long listingId, Long userId, int pageNo, int pageSize);

    Offer createOffer(OfferDto offer);

    Boolean deleteOffer(Long userId, Long offerId);

    Page<Offer> getAllOffers(int pageNo, int pageSize, String sortBy, String search);

}

File: frontend/src/app/services/offers.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Listing} from '../model/Listing';
import {Observable, of} from 'rxjs';
import {PagedOffer} from '../model/PagedOffer';
import {Offer} from '../model/Offer';
import {PagedListing} from '../model/PagedListing';

@Injectable({
  providedIn: 'root'
})
export class OffersService {

  constructor(private http: HttpClient) { }

    public getOffersByOglasId(listingId: number): Observable<PagedOffer> {
        return this.http.get<PagedOffer>(`/api/offers/oglas/${listingId}`);
    }

    public getPersonalOffersByOglasId(listingId: number): Observable<PagedOffer> {
        return this.http.get<PagedOffer>(`/api/offers/oglas/${listingId}/personal`);
    }

    public createNewOffer(offer: any): Observable<Offer> {
      return this.http.post<Offer>('/api/offers', offer, { responseType: 'json'});
    }

    public getAllOffers(
        pageNo: number = 0,
        pageSize: number = 10,
        searchQuery: string = '',
        sortBy: string = ''): Observable<PagedOffer> {
        return this.http.get<PagedOffer>(`/api/offers?pageNo=${pageNo}&pageSize=${pageSize}&search=${searchQuery}&sortBy=${sortBy}`);
    }

    public removeOfferById(offerId: number) {
      return this.http.delete('/api/offers/delete/' + offerId);
    }

}
